### PR TITLE
Calculate the progress component on the backend

### LINKED ISSUE #935: Implement progress bar on item deposit page
Design doc: https://docs.google.com/document/d/1fcr2DYo7OrX-qTdeUOTWKSS1rlq_47sehESmbvgUsrk/edit#heading=h.2j7a9yfzetyj
Wireframes: https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/428088403

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?
Fixes #935 
This makes the progress component on the edit form fully functional and identical to the behavior on the dashboard

### CODE REVIEW

File: app/forms/attached_files_populator.rb [hunk 1]
```diff
@@ -9,15 +9,15 @@ def call(form, fragment:, **)
     item = existing_record(form: form, id: fragment['id'])
 
     if fragment['_destroy'] == '1'
-      # Remove AttachedFile and associated AS model instances if AF exists
-      # Else, there is no AF, so remove the AS::Blob directly
-      if item
-        form.attached_files.delete(item)
-      else
-        ActiveStorage::Blob.find_signed!(fragment['file']).purge_later
```
Comment (mjgiarlo) [inline app/forms/attached_files_populator.rb:17]: Nice to see this go.

File: app/forms/attached_files_populator.rb [hunk 2]
```diff
@@ -9,15 +9,15 @@ def call(form, fragment:, **)
...
-      end
+      # Remove AttachedFile and associated ActiveStorage objects
+      form.attached_files.delete(item) if item
       return skip!
     end
-    item || form.attached_files.append(AttachedFile.new(file: fragment['file']))
+    return item if item
+
+    # When in the ValidationController, sometimes frament['file'] is an empty string. Avoid trying to load that.
```
Comment (mjgiarlo) [inline app/forms/attached_files_populator.rb:18]: Typo: `s/frament/fragment/`

File: app/javascript/controllers/edit_deposit_controller.js [hunk 3]
```diff
@@ -1,36 +1,29 @@
 import { Controller } from "stimulus";
 
 export default class extends Controller {
-  static targets = ["title", "titleField",
```
Comment (mjgiarlo) [inline app/javascript/controllers/edit_deposit_controller.js:4]: :wave:

File: app/javascript/controllers/edit_deposit_controller.js [hunk 4]
```diff
@@ -1,36 +1,29 @@
...
-  }
-
-  checkField(field) {
-    const stepName = field.getAttribute("data-progress-step")
-    const step = this.targets.find(stepName)
-    let isComplete = field.value !== ''
-    if (stepName === 'file') {
-      isComplete = document.querySelectorAll('.dz-preview').length > 0
+    let form = this.element
+    let data = new FormData(form)
```
Comment (mjgiarlo) [inline app/javascript/controllers/edit_deposit_controller.js:16]: Should these be `const`s? :man_shrugging:

File: app/javascript/controllers/edit_deposit_controller.js [hunk 5]
```diff
@@ -1,36 +1,29 @@
...
+      if (key.match(/TEMPLATE_RECORD/)) {
+        data.delete(key)
+      }
     }
-    if (stepName === 'terms') {
-      isComplete = document.querySelector('#work_agree_to_terms').checked
-    }
-    step.classList.toggle('active', isComplete)
+
+    let queryString = new URLSearchParams(data).toString();
```
Comment (mjgiarlo) [inline app/javascript/controllers/edit_deposit_controller.js:25]: `const`?

**Review by mjgiarlo — COMMENTED** [review]
There's a comment typo and some JS niggles, but otherwise good to go.

File: app/forms/attached_files_populator.rb (hunk 1 above)
Comment (jcoyne) [reply app/forms/attached_files_populator.rb:17]: Yes, we don't want a model serializer to be responsible for doing persistence.

File: app/javascript/controllers/edit_deposit_controller.js (hunk 4 above)
Comment (jcoyne) [reply app/javascript/controllers/edit_deposit_controller.js:16]: yep

File: app/javascript/controllers/edit_deposit_controller.js (hunk 5 above)
Comment (jcoyne) [reply app/javascript/controllers/edit_deposit_controller.js:25]: changed.

File: app/forms/attached_files_populator.rb (hunk 2 above)
Comment (jcoyne) [reply app/forms/attached_files_populator.rb:18]: good 👁️

**Review by mjgiarlo — APPROVED** [review]
Thanks for the tweaks. Merge when :green_circle:!

**Pull request merged**